btree/btree.c

```diff
@@ -2752,23 +2791,28 @@ static int64_t m0_btree_put_makespace_phase(struct m0_btree_op *bop)
 	node_fix(tgt.s_node, bop->bo_tx);
 
 	/* Initialized new record which will get inserted at parent */
-	node_slot.s_node = curr_level->l_node;
+	node_slot.s_node = lev->l_node;
 	node_slot.s_idx = 0;
 	node_slot.s_rec = temp_rec;
 	node_key(&node_slot);
 	new_rec.r_key = node_slot.s_rec.r_key;
 
 	newvalue      = INTERNAL_NODE_VALUE_SIZE;
-	newv_ptr      = &(curr_level->l_alloc->n_addr);
+	newv_ptr      = &(lev->l_alloc->n_addr);
 	new_rec.r_val = M0_BUFVEC_INIT_BUF(&newv_ptr, &newvsize);
 
+
+
 	temp_rec_1.r_key.k_data   = M0_BUFVEC_INIT_BUF(&p_key_1, &ksize_1);
 	temp_rec_1.r_val          = M0_BUFVEC_INIT_BUF(&p_val_1, &vsize_1);
 
 	for (i = oi->i_used - 1; i >= 0; i--) {
-		curr_level = &oi->i_level[i];
-		node_slot.s_node = curr_level->l_node;
-		node_slot.s_idx  = curr_level->l_idx;
+		node_put(lev->l_alloc);
+		lev->l_alloc = NULL;
+
+		lev = &oi->i_level[i];
```

Comment:
Should `lev = &oi->i_level[i]` be at line 2810 ?

Reply:
lev at line no. 2810, indicate the level for l_alloc , lev at line no.:2813 indicate it's parents level, once we done splitting, etc, and done with using l_alloc , we want to do node_put and make it as NULL and then we will go level about it and perform operation at it s parent